<?xml version="1.0" ?>
<launch>
  <arg name="t1_limits" 		default="false"/>
  <arg name="controllers" 	default="joint_controllers"/>
  <arg name="start_controller"	default="joint_controllers"/>

  <arg name="rviz_config" default="$(find lwr_launch)/launch/rviz_config.rviz"/>
  <arg name="load-rviz" default="true"/>
  <arg name="rviz_bringup" default="$(arg load-rviz)"/> <!--alias-->

  <arg name="controller_file" 	default="$(find single_lwr_robot)/config/controllers.yaml"/>
  <arg name="hw_interface_file"   default="$(find single_lwr_robot)/config/hw_interface.yaml"/>
  <arg name="joint_limit_file"    default="$(find single_lwr_robot)/config/t1_joint_limits.yaml"/>
  <arg name="fri_driver_file"     default="$(find single_lwr_robot)/config/980241-FRI-Driver.init"/>
  <arg name="robot_urdf_file" 	default="$(find single_lwr_robot)/robot/single_lwr_robot.urdf.xacro"/>

  <!-- Load updated joint limits (override information from single_lwr_moveit) to respect T1 mode limits -->
  <group if="$(arg t1_limits)" ns="robot_description_planning">
    <rosparam file="$(arg joint_limit_file)" command="load"/>
  </group>

  <param name="start_controller" type="string" value="$(arg start_controller)" />

  <!-- load robotHW configurations to rosparam server -->
  <rosparam file="$(arg hw_interface_file)" command="load"/>

  <!-- load all controller configurations to rosparam server -->
  <rosparam file="$(arg controller_file)" command="load"/>

  <param name="fri_drivers" type="str" value="$(arg fri_driver_file)"/>

  <group ns="lwr">
    <param name="robot_description" command="$(find xacro)/xacro.py $(arg robot_urdf_file)"/>

    <node  name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <node  name="controller_spawner_js" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="kuka_joint_state_controller"/>
    <node  name="controller_spawner"    pkg="controller_manager"    type="spawner" respawn="false" output="screen" args="--stopped $(arg controllers)"/>

    <node name="lwr_fri" pkg="lwr_fri" type="lwr_fri_node" output="screen"/>
  </group>

  <!-- LAUNCH RVIZ -->
  <group if="$(arg rviz_bringup)">
    <node name="lwr_rviz" pkg="rviz" type="rviz" respawn="false" args="-d $(arg rviz_config)" output="screen"/>
  </group>
</launch>
